src/components/fx/hover.js

```diff
@@ -689,7 +689,7 @@ function createHoverText(hoverData, opts, gd) {
             // strip out our pseudo-html elements from d.name (if it exists at all)
             name = svgTextUtils.plainText(d.name || '');
 
-            if(name.length > 15) name = name.substr(0, 12) + '...';
+            if(name.length > commonLabelOpts.namelength && commonLabelOpts.namelength > 0) name = name.substr(0, commonLabelOpts.namelength - 3) + '...';
```

Comment:
@etpinard my understanding from this [comment](https://github.com/plotly/plotly.js/issues/1777#issuecomment-308321146) is that when `namelength = 0` there should be no trimming happening, instead display the whole text... Am I wrong?